java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture10/tasks/_1/FileRedactor.java

```diff
@@ -0,0 +1,61 @@
+package com.zoolatech.lecture10.tasks._1;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+final class FileRedactor {
+
+    private final File file;
+
+    public FileRedactor(File file) {
+        this.file = file;
+    }
+
+    public void showFiles() {
+        File[] files = file.listFiles();
+        int counter = 0;
+
+        assert files != null;
+        for (File value : files) {
+            if (value.isFile()) {
+                System.out.println("File number " + ++counter + " : " + value.getName());
+            }
+        }
+    }
+
+    public void generateFiles(String fileName) throws IOException {
+        File[] files = file.listFiles();
+        if (files == null) {
+            System.err.println("File is empty");
+        } else {
+            for (File fileInFiles : files) {
+                if (fileInFiles.exists() && fileInFiles.getName().equals(fileName)) {
+                    try (FileInputStream fileInputStream = new FileInputStream(file + "/" + fileName)) {
+                        String result = new String(fileInputStream.readAllBytes());
+                        System.out.println(result);
+                        fileInputStream.close();
+                        System.out.println("1. Start\n2. Exit");
+                    }
+                } else {
+                    logError(fileName);
```

Comment:
Let's say you have 3 files. If you select the third one it means that before the file will be found the user will see a few error messages and a few times error will be logged. This part can be optimized and we need to log error only if listFiles doesn't have the desired file.